### ISSUE TITLE
Grafana theme for color blinds

### ISSUE THREAD

**Issue by hassanfarid** [grafana/grafana#18041]
**Environment**:
- Grafana version: v6.2

I received this request from one of my internal team members, where we use Grafana.

```
I was looking at grafana dashboards recently and having a lot of difficulty reading the graphs because I have a mild form of red / green color blindness (deuteranomaly, rather than deuteranopia). Basically, when I look at the graphs, all the lines look the same. If it’s not much trouble, for a future update, would you consider adding the option to change the colors?

If you don’t want to add a whole color choosing option, there’s a short set of rules you can use to choose color palettes for people with color problems. I think I can remember them pretty well:
1.	Make palettes available that are:
a.	Blue or violet alongside red, with yellow or a bright orange
b.	Red alongside green (which you already have, for a rare form of color blindness)
2.	Use different saturations and brightnesses for different data. This lets users with even complete color blindness to distinguish differences between the data based on color value.
3.	Using patterns is helpful if that’s an option.

Thanks for considering my request!
```

Can we have grafana theme that make lives of people easier while using grafana? I am not an expert and this field, but I can certainly arrange with some of the people I know to contribute to this topic with information about, what will be helpful to build this. Like what color and pattern strategy can be helpful.

I certainly see that series color can be customized, but how to set default series colors that will work for these scenarios?

**Comment by torkelo** [comment]
You can change the colors for each series using the color picker

For Graphs I am not sure what the solution is for this, we could have a color blind mode and ignore all user defined colors and only use colors that have good contrast for the semi color blind

**Comment by hassanfarid** [comment]
@torkelo - Having color blind mode sounds like a scale-able plan. The objective is to make it enjoyable experience for all users really.

Will this require adding another theme along with light and dark mode?

**Comment by torkelo** [comment]
I don’t think we can have a default theme that accommodates both, Grafana is at its fundamentals a colorful and graphic tool, where a vibrant look is important

**Comment by hassanfarid** [comment]
@torkelo - we don't have to make it default theme. But can we make it as available theme / mode so that can set it as their personal default theme under user preferences?

![image](https://user-images.githubusercontent.com/782236/60989392-498e8700-a31c-11e9-9ad3-0db4c5ca0ac5.png)


The impact of choosing this theme / mode will effect the colors scheme shown in all graphs and dashboard. Somehow overriding the colors selected for normal user case.

**Comment by ryantxu** [comment]
We could add a 'high contrast' theme, that would solve *some* of the problem, but for better accessibility the graph editors need to make good choices about what to show.  In particular graphs with many series colors are difficult to understand.  

See:
https://material.io/design/communication/data-visualization.html#style

![image](https://user-images.githubusercontent.com/705951/60989774-a168c600-a2fb-11e9-8be7-f4ada82404f1.png)

![image](https://user-images.githubusercontent.com/705951/60989849-da089f80-a2fb-11e9-81eb-7fd29ce042cc.png)

Also good to keep in mind:
https://material.io/design/usability/accessibility.html#color-contrast

**Comment by ryantxu** [comment]
See also #13014

**Comment by hassanfarid** [comment]
Has there any development on this request? Is it possible to support this as theme option, which will override page background color and graph points to high contrast values?

**Comment by Routhinator** [comment]
@torkelo 
> You can change the colors for each series using the color picker 

So the problem with this, and how I ended up here - is Heatmaps. You cannot pick colours - and _NONE_ of the many schemes available are red-green colour blind friendly. 

Red-Green colour blindness does best with Blue and Orange. The closest we can get to this in Heatmaps right now is Purple/Orange, and Purple is not great due to it containing red - plus the exact shades/hues chosen by the Grafana team are not great.

> Is this still relevant? If so, what is blocking it? Is there anything you can do to help move it forward?

Absolutely still relevant until the above problem is addressed.

**Comment by yaakovfeldman** [comment]
Colour blindness affects perhaps 5% of people, and most find it hard to distinguish the red and green series. This could be 90% solved by https://github.com/grafana/grafana/issues/17092 and allowing a choice of palettes. There are many well known excellent safe choices in the data visualisation community for both discrete and continuous data eg viridis.

**Comment by LarsStegman** [comment]
Not stale

**Comment by mikehearn** [comment]
It's actually more than 5% in practice for a tool like this, because the user base will be heavily gender-skewed. For men about 8% of us are colour blind. The risk with non-colourblind defaults is that someone will quickly glance at a dashboard and draw the wrong conclusion. When I did SRE work I used to have to spend the first day adding colourblind friendly modes to everything (of course, that could also be the default, but usually people feel CBF palettes are ugly).

**Comment by ashharrison90** [comment]
just a small update:
- we added some standard colorblind-friendly viz palettes with [FieldColor: Add accessible color palettes](https://github.com/grafana/grafana/pull/114424#top)
- we're aiming to add some overall colorblind-friendly UI themes as part of [Add Accessible Themes](https://github.com/grafana/grafana/issues/113370) soon™️ 
  - it's likely these will only tackle the main UI colors (primary/secondary/success/warning/error/info)
  - because of the existing customisation options in visualisations (there are 30 different colors in the color picker...), it's very hard/impossible to modify those colors in a way that will work for people with color vision deficiencies 
  - the recommendation here would be to use one of the newly added colorblind-friendly palettes. see screenshots below

<img width="318" height="384" alt="Image" src="https://github.com/user-attachments/assets/a3495df9-d4e3-4428-b4ea-0099f440ceb9" />

<img width="1900" height="753" alt="Image" src="https://github.com/user-attachments/assets/729cc058-a226-4492-bae3-50aa234d221c" />

**Comment by mikehearn** [comment]
Looks great! I mean, it looks psychedelic but that's what we need, thanks :)
